frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/FolderItem.hooks.ts

```diff
@@ -30,58 +56,130 @@ export const useFolderEvents = (folder: IFolder) => {
     setShowRenamePopup(true);
   };
 
-  const changeFolderTitle = (newTitle: string) => {
+  const changeFolderTitle = async (newTitle: string) => {
+    await FolderEventUpdateApp(
+      UpdateAppPayloadPB.fromObject({
+        name: newTitle,
+        desc: '',
+        app_id: folder.id,
+      })
+    );
     appDispatch(foldersActions.renameFolder({ id: folder.id, newTitle }));
   };
 
   const closeRenamePopup = () => {
     setShowRenamePopup(false);
   };
 
-  const deleteFolder = () => {
+  const deleteFolder = async () => {
     closePopup();
+    await FolderEventDeleteApp(
+      AppIdPB.fromObject({
+        value: folder.id,
+      })
+    );
     appDispatch(foldersActions.deleteFolder({ id: folder.id }));
   };
 
-  const duplicateFolder = () => {
+  const duplicateFolder = async () => {
     closePopup();
-    appDispatch(foldersActions.addFolder({ id: nanoid(8), title: folder.title }));
+    const createAppResult = await FolderEventCreateApp(
+      CreateAppPayloadPB.fromObject({
+        workspace_id: workspace.id,
+        name: folder.title,
+        desc: '',
+      })
+    );
+    if (createAppResult.ok) {
+      const pb = createAppResult.val;
+      appDispatch(foldersActions.addFolder({ id: pb.id, title: folder.title }));
+    } else {
+      throw new Error('create folder error');
+    }
   };
 
   const closePopup = () => {
     setShowFolderOptions(false);
     setShowNewPageOptions(false);
   };
 
-  const onAddNewDocumentPage = () => {
+  const onAddNewDocumentPage = async () => {
     closePopup();
-    appDispatch(
-      pagesActions.addPage({
-        folderId: folder.id,
-        pageType: ViewLayoutTypePB.Document,
-        title: 'New Page 1',
-        id: nanoid(6),
+    const createViewResult = await FolderEventCreateView(
+      CreateViewPayloadPB.fromObject({
+        name: 'New Page 1',
+        layout: ViewLayoutTypePB.Document,
+        belong_to_id: folder.id,
+        data_format: ViewDataFormatPB.NodeFormat,
+        desc: '',
+        thumbnail: '',
+        initial_data: new Uint8Array([]),
       })
     );
+    if (createViewResult.ok) {
+      const pb = createViewResult.val;
+      appDispatch(
+        pagesActions.addPage({
+          folderId: folder.id,
+          pageType: ViewLayoutTypePB.Document,
+          title: pb.name,
+          id: pb.id,
+        })
+      );
+    } else {
+      throw new Error('create view error');
+    }
   };
 
-  const onAddNewBoardPage = () => {
+  const onAddNewBoardPage = async () => {
     closePopup();
-    appDispatch(
-      pagesActions.addPage({
-        folderId: folder.id,
-        pageType: ViewLayoutTypePB.Board,
-        title: 'New Board 1',
-        id: nanoid(6),
+    const createViewResult = await FolderEventCreateView(
```

Comment:
Is the call chain like this?
appDispatch -> action -> handler (interact with the backend using AppBackendService) .

Reply:
refactored to use backend service